src/main/java/com/adobe/aem/compgenerator/web/ChildPropBuilderServlet.java

```diff
@@ -0,0 +1,321 @@
+        }
+        if (reConfig.has("type") && !reConfig.get("type").isNull()) {
+            JsonNode type = reConfig.get("type");
+            if (type.has("value")) {
+                String value = type.get("value").textValue();
+                property.setType(value);
+            }
+        }
+        if (reConfig.has("attributes")) {
+            Map<String, String> updatedAttributes = new HashMap<>();
+            JsonNode attrArr = reConfig.get("attributes");
+            if (attrArr.isArray()) {
+                for (final JsonNode objNode : attrArr) {
+                    updatedAttributes.put(objNode.get("key").asText(), objNode.get("value").asText());
+                }
+            }
+            property.setAttributes(updatedAttributes);
+        }
+    }
+
+    private int getIndexOfPropertyById(List<Property> properties, String id) {
+        int index = -1;
+        for (int i = 0; i < properties.size(); i++) {
+            if (properties.get(i).getId().equals(id)) {
+                index = i;
+                break;
+            }
+        }
+        return index;
+    }
+
+    private List<Property> getExistingChildPropertyFromConfig(List<Property> properties, String id, String parentId) {
+        List<Property> existingProperty = properties
+                .stream()
+                .filter(property -> property.getId().equals(parentId))
+                .collect(Collectors.toList());
+        if (existingProperty.isEmpty()) {
+            return new ArrayList<>();
+        } else {
+            if (Objects.isNull(existingProperty.get(0).getItems())) {
+                return new ArrayList<>();
+            } else {
+                return existingProperty.get(0).getItems()
+                        .stream()
+                        .filter(property -> property.getId().equals(id))
+                        .collect(Collectors.toList());
+            }
+        }
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        resp.setContentType("application/json; charset=UTF-8");
+        PrintWriter writer = resp.getWriter();
+        ObjectMapper mapper = new ObjectMapper();
+        Message msg = new Message(true, "Child property builder getter");
+        writer.write(mapper.writeValueAsString(msg));
+        writer.close();
+    }
+
+    @Override
+    /*
+     * Handles updates to the child items of dialog properties
+     */
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        resp.setContentType("application/json; charset=UTF-8");
+        resp.setCharacterEncoding("UTF-8");
+        ObjectMapper mapper = new ObjectMapper();
+        File configFile = new File(CONFIG_PATH);
+        // sanity check for config file existence
+        if (!configFile.exists()) {
+            resp.setStatus(500);
+            PrintWriter writer = resp.getWriter();
```

Comment:
It's not a big deal, but even putting some of this repeated code (get writer, write message, close writer) into a util function will remove quite a bit of code from this and the other similar servlets.

Reply:
Good idea. will update.